node/service_setup.go

```diff
@@ -93,7 +93,7 @@ func (node *Node) setupForClientNode() {
 	nodeConfig, chanPeer := node.initNodeConfiguration()
 
 	// Register peer discovery service.
-	node.serviceManager.RegisterService(service.PeerDiscovery, discovery.New(node.host, nodeConfig, chanPeer))
+	node.serviceManager.RegisterService(service.PeerDiscovery, discovery.New(node.host, nodeConfig, chanPeer, node.AddBeaconPeer))
```

Comment:
ClientNode doesn't need doesn't need add beacon peers, since we already have function to receive beacon block. But maybe we will change this later for client node.

Reply:
will pass in a nil pointer here.